lualua.c

```diff
@@ -264,16 +263,42 @@ static int lualua_newuserdata(lua_State *L) {
   return 1;
 }
 
+typedef struct {
+  lua_State *state;
+  int nargs;
+  int nresults;
+} lualua_Call;
+
+static int lualua_docall(lua_State *L) {
+  lualua_Call *call = lua_touserdata(L, 1);
+  lua_call(call->state, call->nargs, call->nresults);
+  return 0;
+}
+
 static int lualua_pcall(lua_State *L) {
   lualua_State *S = lualua_checkstate(L, 1);
   int nargs = luaL_checkint(L, 2);
   int nresults = luaL_checkint(L, 3);
   int errfunc = luaL_checkint(L, 4);
-  lualua_assert(S, errfunc == 0 || lualua_isacceptableindex(S, errfunc),
-                "invalid index");
+  if (errfunc != 0) {
```

Comment:
Obvious this restriction needs to be lifted, but before I take that on, want feedback on the general approach.